src/main/scala/dpla/ingestion3/enrichments/SpatialEnrichment.scala

```
@@ -0,0 +1,275 @@
+    * @return   The coordinates as at "lat, lng" string
+    */
+  private def coordinates(j: JValue): Option[String] = {
+    val c = ((((j \ "interpretations")(0) \ "feature") \ "geometry") \
+      "center").extract[Coords]
+    Option(c.lat.toString + "," + c.lng.toString)
+  }
+
+  /**
+    * Return the place name or None if it's the place's coordinates
+    *
+    * If the place's name field has been used to convey coordinates, we do want
+    * to override this when we fill it in.  Otherwise, we don't want to clobber
+    * it if it's already defined.
+    *
+    * @param nameCandidate
+    * @param j
+    */
+  private def nameUnlessCoords(nameCandidate: Option[String], j: JValue):
+      Option[String] = {
+    nameCandidate match {
+      case s if s.isDefined => {
+        if (s.mkString matches """[\+\-\d\.]+\s*,\s*[\+\-\d\.]+""") {
+          // It's coordinates, so return the name instead
+          displayName(j)
+        } else {
+          s
+        }
+      }
+      case _ => None
+    }
+  }
+
+  /**
+    * Enrich the given DplaPlace
+    *
+    * @param p  The Place
+    * @return   An enriched copy of the Place
+    */
+  def enrich(p: DplaPlace): DplaPlace = {
+    try {
+      queryTerm(p) match {
+        case Some(term) => {
+          val j = geocoder.geocoderResponse(term)
+          // Return the looked-up place, unless it was a country.
+          if (woeType(j) != Some(12)) {
+            p.copy(
+              name = nameUnlessCoords(p.name, j),
+              city = p.city.orElse(city(j)),
+              county = p.county.orElse(county(j)),
+              state = p.state.orElse(state(j)),
+              country = p.country.orElse(country(j)),
+              coordinates = p.coordinates.orElse(coordinates(j))
+            )
+          } else {
+            p
+          }
+        }
+        case _ => p
+      }
+    } catch {
+      case e: Exception => {
+        logger.error(e.toString)  // TODO: Catch this higher up and store it?
```

If you let this exception propagate, it'll end up at the driver for the operator to see, actually. If you only log it, you have to go to a log file/system to find out that anything is wrong.

At the very least, you should rethrow the exception with `throw e` I think.